Minor Updates

RHDZMOTA opened the pull request:
This PR contains the implementation of the [SemVer Specification](https://semver.org/) (fixes #3 ) and some minor refactoring. 

Run tests: `sbt test`

RHDZMOTA committed:
add akka http jackson

RHDZMOTA committed:
Change to semantic versioning

RHDZMOTA committed:
Update tests for SemVer

RHDZMOTA committed:
insignificant refactor

RHDZMOTA committed:
Update version

RHDZMOTA committed:
Update build according to SemVer

RHDZMOTA committed:
Update README

Rafailong commented on README.md:
```diff
@@ -4,7 +4,7 @@ This is an Akka HTTP hello-world example compatible with Heroku.
 ## Usage
 
-To run locally (at `0.0.0.0:8080/v1/hello`):
+To run locally (at `0.0.0.0:8080/v1.0.0/hello`):
```
As you are adding API versioning schema, I think it'd be good to include in here some information 'bout it. I mean, generals of the versioning schema.

RHDZMOTA replied:
Good idea, I will specify the project and API version in the top of the README as well as the schema they follow.

Rafailong commented on build.sbt:
```diff
@@ -7,17 +7,14 @@ lazy val root = (project in file(".")).
       organization := "com.rhdzmota",
       scalaVersion := "2.12.6",
-      version      := "0.1.0-SNAPSHOT"
+      version      := "1.0.0"
```
API versioning and project versions are different. You'd better not mixing them.

RHDZMOTA replied:
That's correct. Following your comment on using **ordinal numbers** for the API versioning, I think that the project version can stay with the **semver specification**.

Rafailong commented:
I did notice that this PR includes at least 2 big refactors:

1. API versioning schema, and
2. Languages refactor.

Take into account that PRs like this one can be hard to review as they do not isolate one thing only. As a better practice try to send a PR per issue/feature/refactor.

P.D. PRs about code styling must be not contain business logic refactoring.

Rafailong commented on src/test/scala/com/rhdzmota/examples/ServerSpecs.scala:
```diff
@@ -1,57 +1,57 @@
-    "return a hello-world json (english) for GET request to /v1/hello" in {
-      Get("/v1/hello") ~> Server.route ~> check {
+    "return a hello-world json (english) for GET request to /v1.0.0/hello" in {
+      Get("/v1.0.0/hello") ~> Server.route ~> check {
```
Would it be a good idea to include a package object to isolate constants like the API version?

RHDZMOTA replied:
Good idea, haven't used package objects before. I'll add one for this use case following [this guideline](https://alvinalexander.com/scala/scala-package-objects-how-to-name-location-cookbook-recipe).  Thanks for the recommendation.

Rafailong reviewed (commented):
For API versioning:
1. Use a simple ordinal number.
2. Don't use the dot notation like `v1.0.1`, due to it implies a granularity of versioning that does not work well with APIs which are interfaces not implementations.

RHDZMOTA committed:
Add package object with API versioning

Create a package object that contains the versioning history and the
current version.

RHDZMOTA committed:
Optional Path Prefix with version

Create a versionless route (permalink) and an optional version path (v1).
Future releases should contemplate to inform the user if they are using
old versions (e.g. v0) with Http status codes (301 Moved permanently).

RHDZMOTA committed:
Test versionless path

Perform tests on the current version of the API.

RHDZMOTA committed:
Add Version Schema

Specify project and API versions with their corresponding schema.
